--- src/nm-device-802-11-wireless.c	2006-06-20 20:15:45.000000000 +0200
+++ src/nm-device-802-11-wireless.c.new	2006-08-13 22:32:13.000000000 +0200
@@ -2329,7 +2329,7 @@
 }
 
 
-#define NM_SUPPLICANT_TIMEOUT	20	/* how long we wait for wpa_supplicant to associate (in seconds) */
+#define NM_SUPPLICANT_TIMEOUT	60	/* how long we wait for wpa_supplicant to associate (in seconds) */
 
 static unsigned int
 get_supplicant_timeout (NMDevice80211Wireless *self)
