--- NetworkManagerDevice.c	2005-10-19 16:19:53.000000000 +0200
+++ src/NetworkManagerDevice.c	2006-01-11 12:14:53.000000000 +0100
@@ -4283,7 +4283,7 @@
 
 	strncpy (ifr.ifr_name, nm_device_get_iface (dev), sizeof(ifr.ifr_name)-1);
 	edata.cmd = ETHTOOL_GLINK;
-	ifr.ifr_data = (char *) &edata;
+	ifr.ifr_data = (void __user *) &edata;
 #ifdef IOCTL_DEBUG
 	nm_info ("%s: About to ETHTOOL\n", nm_device_get_iface (dev));
 #endif
@@ -4305,8 +4305,9 @@
 /**************************************/
 /*    MII capability detection        */
 /**************************************/
-#include <linux/mii.h>
-
+#define _LINUX_IF_H
+	#include <linux/mii.h>
+#undef _LINUX_IF_
 static int mdio_read (NMDevice *dev, NMSock *sk, struct ifreq *ifr, int location)
 {
 	struct mii_ioctl_data *mii;
