Index: src/NetworkManager.conf
===================================================================
--- src/NetworkManager.conf	(Revision 3269)
+++ src/NetworkManager.conf	(Arbeitskopie)
@@ -11,6 +11,14 @@
                 <allow send_destination="org.freedesktop.NetworkManager.PPP"/>
                 <allow send_interface="org.freedesktop.NetworkManager.PPP"/>
         </policy>
+        <policy user="haldaemon">
+                <allow send_destination="org.freedesktop.NetworkManager"/>
+                <allow send_interface="org.freedesktop.NetworkManager"/>
+        </policy>
+        <policy group="netdev">
+                <allow send_destination="org.freedesktop.NetworkManager"/>
+                <allow send_interface="org.freedesktop.NetworkManager"/>
+        </policy>
         <policy at_console="true">
                 <allow send_destination="org.freedesktop.NetworkManager"/>
                 <allow send_interface="org.freedesktop.NetworkManager"/>
