Index: NetworkManager/system-settings/src/org.freedesktop.NetworkManagerSystemSettings.service.in
===================================================================
--- NetworkManager.orig/system-settings/src/org.freedesktop.NetworkManagerSystemSettings.service.in	2008-06-11 12:28:11.000000000 +0200
+++ NetworkManager/system-settings/src/org.freedesktop.NetworkManagerSystemSettings.service.in	2008-06-11 12:32:51.000000000 +0200
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=org.freedesktop.NetworkManagerSystemSettings
-Exec=@sbindir@/nm-system-settings --config /etc/nm-system-settings.conf
+Exec=@sbindir@/nm-system-settings --config /etc/NetworkManager/nm-system-settings.conf
 User=root
 
