--- src/nm-device-802-3-ethernet.c	2006-05-13 21:06:05.000000000 +0200
+++ src/nm-device-802-3-ethernet.c.new	2006-08-13 22:34:56.000000000 +0200
@@ -376,7 +376,9 @@
 /**************************************/
 /*    MII capability detection        */
 /**************************************/
+#define _LINUX_IF_H
 #include <linux/mii.h>
+#undef _LINUX_IF_H
 
 static int
 mdio_read (NMDevice8023Ethernet *self, NMSock *sk, struct ifreq *ifr, int location)
