Index: NetworkManager-0.7.999/system-settings/plugins/ifupdown/plugin.c
===================================================================
--- NetworkManager-0.7.999.orig/system-settings/plugins/ifupdown/plugin.c	2010-02-08 16:07:48.379559508 +0100
+++ NetworkManager-0.7.999/system-settings/plugins/ifupdown/plugin.c	2010-02-08 16:08:04.255557965 +0100
@@ -355,7 +355,7 @@
 	ifparser_init ();
 	block = ifparser_getfirst ();
 	while (block) {
-		if(!strcmp ("auto", block->type))
+		if(!strcmp ("auto", block->type) || !strcmp ("allow-hotplug", block->type))
 			g_hash_table_insert (auto_ifaces, block->name, GUINT_TO_POINTER (1));
 		else if (!strcmp ("iface", block->type) && strcmp ("lo", block->name)) {
 			NMIfupdownConnection *exported;
