--- src/backends/NetworkManagerDebian.c	2006-02-28 22:24:22.000000000 +0100
+++ ../NetworkManager-0.6.0.patched/src/backends/NetworkManagerDebian.c	2006-03-06 12:19:34.000000000 +0100
@@ -37,6 +37,7 @@
 #include "nm-utils.h"
 
 #define ARPING "/usr/sbin/arping"
+#define RESOLVCONF "resolvconf"
 
 /*
  * nm_system_init
@@ -679,5 +680,8 @@
  */
 gboolean nm_system_should_modify_resolv_conf (void)
 {
-	return TRUE;
+	if (g_find_program_in_path(RESOLVCONF) != NULL)
+		return FALSE;
+	else
+		return TRUE;
 }
