commit ab7e3a44e2869b91a62d12b3157fa1debc5df7a9
Author: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date:   Sun Jun 22 22:23:38 2008 +0100

    Execute system dhclient script

diff --git a/callouts/nm-dhcp-client-action.c b/callouts/nm-dhcp-client-action.c
index 9917235..30d7a02 100644
--- a/callouts/nm-dhcp-client-action.c
+++ b/callouts/nm-dhcp-client-action.c
@@ -34,6 +34,9 @@
 #define NM_DHCP_CLIENT_DBUS_SERVICE "org.freedesktop.nm_dhcp_client"
 #define NM_DHCP_CLIENT_DBUS_IFACE   "org.freedesktop.nm_dhcp_client"
 
+
+#define SYSTEM_DHCLIENT_SCRIPT "/sbin/dhclient-script"
+
 /**
  * Start a dict in a dbus message.  Should be paired with a call to
  * {@link wpa_dbus_dict_close_write}.
@@ -295,6 +298,8 @@ main (int argc, char *argv[])
 
 	g_type_init ();
 
+	g_spawn_command_line_sync (SYSTEM_DHCLIENT_SCRIPT, NULL, NULL, NULL, NULL);
+
 	/* Get a connection to the system bus */
 	connection = dbus_init ();
 	if (connection == NULL)
