Index: man/nm-tool.1.in
===================================================================
RCS file: /cvs/gnome/NetworkManager/man/nm-tool.1.in,v
retrieving revision 1.1
diff -u -3 -p -r1.1 nm-tool.1.in
--- man/nm-tool.1.in	1 Feb 2006 18:14:15 -0000	1.1
+++ man/nm-tool.1.in	30 Nov 2006 19:33:39 -0000
@@ -3,6 +3,7 @@
 .\" Copyright (C) 2005 Robert Love
 .\"
 .TH NM-TOOL "1"
+.SH NAME
 nm-tool \- utility to report NetworkManager state 
 .SH SYNOPSIS
 .B nm-tool
