Index: src/NetworkManagerAP.c
===================================================================
--- src/NetworkManagerAP.c	(Revision 2356)
+++ src/NetworkManagerAP.c	(Revision 2357)
@@ -572,12 +572,14 @@
 {
 	if (cipher & IW_AUTH_CIPHER_WEP40)
 	{
-		caps |= (NM_802_11_CAP_PROTO_WEP & NM_802_11_CAP_CIPHER_WEP40);
+		caps |= NM_802_11_CAP_PROTO_WEP;
+		caps |= NM_802_11_CAP_CIPHER_WEP40;
 		caps &= ~NM_802_11_CAP_PROTO_NONE;
 	}
 	if (cipher & IW_AUTH_CIPHER_WEP104)
 	{
-		caps |= (NM_802_11_CAP_PROTO_WEP & NM_802_11_CAP_CIPHER_WEP104);
+		caps |= NM_802_11_CAP_PROTO_WEP;
+		caps |= NM_802_11_CAP_CIPHER_WEP104;
 		caps &= ~NM_802_11_CAP_PROTO_NONE;
 	}
 	if (cipher & IW_AUTH_CIPHER_TKIP)
