Index: ChangeLog
===================================================================
--- ChangeLog	(Revision 2753)
+++ ChangeLog	(Revision 2754)
@@ -1,3 +1,8 @@
+2007-08-30  Dan Williams  <dcbw@redhat.com>
+
+	* libnm-util/sha1.c
+		- Include config.h to get defines for endiannes (gnome.org #420216)
+
 2007-07-26  Dan Williams  <dcbw@redhat.com>
 
 	Patch from Bernhard Miklautz <bernhard.miklautz@shacknet.at>
Index: libnm-util/sha1.c
===================================================================
--- libnm-util/sha1.c	(Revision 2753)
+++ libnm-util/sha1.c	(Revision 2754)
@@ -12,6 +12,10 @@
  * See README and COPYING for more details.
  */
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 
